src/ddb/ddb_cmd_options.c

```diff
@@ -219,38 +282,195 @@ rm_option_parse(struct ddb_ctx *ctx, struct rm_options *cmd_args, struct argv_pa
+{
+	char		 *options_short = "";
+	int		  index = 0, opt;
+	uint32_t	  argc = argc_v->ap_argc;
+	char		**argv = argc_v->ap_argv;
+	struct option	  options_long[] = {
+		{ NULL }
+	};
+
+	memset(cmd_args, 0, sizeof(*cmd_args));
+
+	/* Restart getopt */
+	optind = 1;
+	opterr = 0;
+	while ((opt = getopt_long(argc, argv, options_short, options_long, &index)) != -1) {
+		switch (opt) {
+		case '?':
+			ddb_printf(ctx, "Unknown option: '%c'\n", optopt);
+		default:
+			return -DER_INVAL;
+		}
+	}
+
+	index = optind;
+
+	D_ASSERT(argc > index);
+	D_ASSERT(same(argv[index], COMMAND_NAME_RM_ILOG));
+	index++;
+	if (argc - index > 0) {
+		cmd_args->path = argv[index];
 		index++;
 	} else {
-		ddb_print(ctx, "Expected argument 'dst'");
+		ddb_print(ctx, "Expected argument 'path'");
 		return -DER_INVAL;
 	}
+
 	if (argc - index > 0) {
-		cmd_args->epoch = argv[index];
+		ddb_printf(ctx, "Unexpected argument: %s\\n", argv[index]);
+		return -DER_INVAL;
+	}
+
+	return 0;
+}
+
+/* Parse command line options for the 'dump_dtx' command */
+static int
+dump_dtx_option_parse(struct ddb_ctx *ctx, struct dump_dtx_options *cmd_args,
+		      struct argv_parsed *argc_v)
+{
+	char		 *options_short = "ac";
+	int		  index = 0, opt;
+	uint32_t	  argc = argc_v->ap_argc;
+	char		**argv = argc_v->ap_argv;
+	struct option	  options_long[] = {
+		{ "active", no_argument, NULL, 'a' },
+		{ "committed", no_argument, NULL, 'c' },
+		{ NULL }
+	};
+
+	memset(cmd_args, 0, sizeof(*cmd_args));
+
+	/* Restart getopt */
+	optind = 1;
+	opterr = 0;
+	while ((opt = getopt_long(argc, argv, options_short, options_long, &index)) != -1) {
+		switch (opt) {
+		case 'a':
+			cmd_args->active = true;
+			break;
+		case 'c':
+			cmd_args->committed = true;
+			break;
+		case '?':
+			ddb_printf(ctx, "Unknown option: '%c'\n", optopt);
+		default:
+			return -DER_INVAL;
+		}
+	}
+
+	index = optind;
+
+	D_ASSERT(argc > index);
+	D_ASSERT(same(argv[index], COMMAND_NAME_DUMP_DTX));
+	index++;
+	if (argc - index > 0) {
+		cmd_args->path = argv[index];
 		index++;
 	} else {
-		ddb_print(ctx, "Expected argument 'epoch'");
+		ddb_print(ctx, "Expected argument 'path'");
+		return -DER_INVAL;
+	}
+
+	if (argc - index > 0) {
+		ddb_printf(ctx, "Unexpected argument: %s\\n", argv[index]);
+		return -DER_INVAL;
+	}
+
+	return 0;
+}
+
+/* Parse command line options for the 'clear_dtx' command */
+static int
+clear_dtx_option_parse(struct ddb_ctx *ctx, struct clear_dtx_options *cmd_args,
+		       struct argv_parsed *argc_v)
+{
+	char		 *options_short = "";
+	int		  index = 0, opt;
+	uint32_t	  argc = argc_v->ap_argc;
+	char		**argv = argc_v->ap_argv;
+	struct option	  options_long[] = {
+		{ NULL }
+	};
+
+	memset(cmd_args, 0, sizeof(*cmd_args));
+
+	/* Restart getopt */
+	optind = 1;
+	opterr = 0;
+	while ((opt = getopt_long(argc, argv, options_short, options_long, &index)) != -1) {
+		switch (opt) {
+		case '?':
+			ddb_printf(ctx, "Unknown option: '%c'\n", optopt);
+		default:
+			return -DER_INVAL;
```

Comment:
Since the option list is empty, then the "switch" will always return -DER_INVAL, is it expected?